js/vector.js

```diff
@@ -0,0 +1,63 @@
+class Vector {
+
+    constructor(x, y) {
+        if (arguments.length === 0) {
+            this.x = 0;
+            this.y = 0;
+        } else {
+            this.x = x;
+            this.y = y;
+        }
+    }
+
+    add(v, y) {
```

Comment:
the method overloading was a bit difficult to understand here. we might want comments or better variable names to understand whats going on

Reply:
also i wonder if this should _return_ a vector instead of updating the vector called. I can imagine sometimes we might just want to calculate a vector sum and not update either one